goalie/mesh_seq.py

```diff
@@ -482,10 +482,17 @@ def _reinitialise_fields(self, initial_conditions):
         """
         for field, initial_condition in initial_conditions.items():
             fs = initial_condition.function_space()
-            self.fields[field] = (
-                firedrake.Function(fs, name=field),
-                firedrake.Function(fs, name=f"{field}_old").assign(initial_condition),
-            )
+            if self.field_types[field] == "steady":
+                self.fields[field] = firedrake.Function(fs, name=f"{field}").assign(
+                    initial_condition
+                )
+            else:
+                self.fields[field] = (
+                    firedrake.Function(fs, name=field),
+                    firedrake.Function(fs, name=f"{field}_old").assign(
+                        initial_condition
+                    ),
+                )
```

Comment:
This looks ugly now but I left it for now since it will all get reworked (soon?) with #181 :) but happy to tidy up if needed